intro-to-R-tidyverse/03-intro_to_r_tidyverse_exercise.Rmd

```diff
@@ -0,0 +1,160 @@
+
+```{r}
+#load tidyverse library
+
+```
+
+#### Importing Files
+Let's import all the files we will need for this exercise. 
+The metadata is present in `exercise-data/metadata_GSE49243.tsv` and gene expression matrix is in `exercise-data/GSE49243.tsv`
+Go ahead and read the metadata and gene expression matrix into variables called `metadata` and `gene_expression`
+
+```{r}
+# Reading Metadata
+metadata <- 
+
+# Reading Gene Expression Matrix
+gene_expression <- 
+```
+
+### Cleaning metadata
+First, we are going to perform some cleaning activities like filtering, renaming, reordering, etc and then write the cleaned metadata into a new file. 
+
+
+#### Inspect the metadata
+Now that you have read the metadata into the variable `metadata`, explore what the metadata looks like. 
+
+```{r}
+# Explore the metadata
+
+```
+
+
+#### Filtering and trimming
+Next, we're going to do some manipulations on the metadata to retain only the information we are interested in. 
+Below we have done some manipulations on the metadata. Take a look and then perform the same manipulations using pipes `%>%` and store the result in `metadata_clean`. 
+
+```{r}
+# Manipulations w/o pipes
+metadata_nopipe <- metadata
+metadata_nopipe <- filter(metadata_nopipe, refinebio_age < 18)
+metadata_nopipe <- select(metadata_nopipe, refinebio_accession_code, refinebio_age,refinebio_sex,histology)
+metadata_nopipe <- arrange(metadata_nopipe, refinebio_age)
+```
+
+It's your turn! 
+
+```{r}
+# Now you try it, with pipes %>%
+metadata_clean <- 
+  
+```
+
+Run the code chunk below to check if you got the same results. 
+
+```{r}
+all.equal(metadata_nopipe, metadata_clean)
+```
+
+#### Rename columns
+The column names `refinebio_accession_code`, `refinebio_age`, `refinebio_sex` are too unwieldy. Let's rename it to something more manageable like `sample_id`, `age`, and `sex` respectively. 
+
+To do this, tidyverse has a function `dplyr::rename`. Use the help section to understand how you can use this function. Store the renamed metadata back into the variable `metadata_clean`. 
+(Feel free to Google for solutions as well!)
+
+```{r}
+# Use the Help section to learn how you can use the rename function
+metadata_clean <- 
+
+```
+
+Use head() function check if the column names changed
+
+```{r}
+
+```
+
+
+### Summarize metadata
+Now, let's calculate the mean age by sex, mean age by histology, and counts of each tissue by sex. 
+
+If you get stuck at any point, feel free to refer to the Help section or google for solutions.
+
+```{r}
+# Calculate mean age by sex
+
+```
+
+
+```{r}
+# Calculate mean age by histology
+
+
+```
+
+
+```{r}
+# Calculate how many of each tissue type exists for each sex.
+
+
+```
+
+
+### Select the samples in the Gene Experssion Matrix
+Now, select the samples in the gene expression matrix so that it has the same samples as `metadata_clean`. (Remember we read the gene expression matrix into `gene_expression`)
```

Comment:
```suggestion
Now, select the samples in the gene expression matrix so that it has the same samples as `metadata_clean`. 
(Remember we read the gene expression matrix into `gene_expression`)
```